www-src/app/replicator/change/change_event_handler.coffee

```diff
@@ -1,28 +1,68 @@
+AndroidCalendarHandler = require "../../lib/android_calendar_handler"
+CozyToAndroidEvent = require "../transformer/cozy_to_android_event"
 log = require('../../lib/persistent_log')
     prefix: "ChangeEventHandler"
     date: true
 
 module.exports = class ChangeEventHandler
 
-    change: (doc) ->
-        log.info "change"
+    constructor: (@calendarSync) ->
+        @androidCalendarHandler = new AndroidCalendarHandler()
+        @cozyToAndroidEvent = new CozyToAndroidEvent()
+        @calendarSync = navigator.calendarsync unless @calendarSync
 
-        if doc._rev.split('-')[0] is 1
-            @create doc
-        else
-            @update doc
+    dispatch: (cozyEvent) ->
+        log.info "dispatch"
 
-    create: (doc) ->
+        @calendarSync.eventBySyncId cozyEvent._id, (err, androidEvents) =>
+            if androidEvents.length > 0
+                androidEvent = androidEvents[0]
+                if cozyEvent._deleted
+                    @delete cozyEvent, androidEvent
+                else
+                    @update cozyEvent, androidEvent
+            else
+                @create cozyEvent unless cozyEvent._deleted
+
+    create: (cozyEvent) ->
         log.info "create"
 
-        console.log doc
+        calendarName = cozyEvent.tags[0]
+        @androidCalendarHandler.getOrCreate calendarName, (err, calendar) =>
+            return log.error err if err
+
+            androidEvent = @cozyToAndroidEvent.transform cozyEvent, calendar
+            @calendarSync.addEvent androidEvent, \
+                    @androidCalendarHandler.ACCOUNT, (err, androidEventId) ->
+                log.error err if err
 
-    update: (doc) ->
+    update: (cozyEvent, androidEvent = undefined) ->
```

Comment:
Why ` androidEvent = undefined ` ?
1. It's just an explicitation of what's actually automatically done by javascript
2. This method can't work without this argument (?)